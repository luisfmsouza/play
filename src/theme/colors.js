export const skyLighter = "#F9FAFB";
export const skyLight = "#F4F6F8";
export const sky = "#DFE3E8";
export const skyDark = "#C4CDD5";
export const inkLightest = "#919EAB";
export const inkLighter = "#637381";
export const inkLight = "#454F5B";
export const ink = "#212B36";
