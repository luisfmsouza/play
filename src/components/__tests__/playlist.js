import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";

import * as colors from "../../theme/colors";

import Playlist from "../playlist";

describe("Playlist component", () => {
  const build = state => {
    const store = {
      getState: () => state,
      dispatch: jest.fn(),
      subscribe: jest.fn()
    };

    return renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={{ colors }}>
            <Playlist store={store} />
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
  };

  describe("when there is playlists", () => {
    it("renders default component", () => {
      const state = {
        playlists: [
          [
            {
              artist: "Artist 1",
              title: "Title 1",
              url: "https://www.youtube.com/embed/MeSu5_Do8P8"
            },
            {
              artist: "Artist 2",
              title: "Title 2",
              url: "https://www.youtube.com/embed/249i5jNRClU"
            }
          ],
          [],
          [
            {
              artist: "Artist 3",
              title: "Title 3",
              url: "https://www.youtube.com/embed/qSeaR7Dqvdc"
            },
            {
              artist: "Artist 4",
              title: "Title 4",
              url: "https://www.youtube.com/embed/Dw5orx_47uY"
            },
            {
              artist: "Artist 5",
              title: "Title 5",
              url: "https://www.youtube.com/embed/sNFndSeEt0w"
            }
          ],
          [
            {
              artist: "Artist 6",
              title: "Title 6",
              url: "https://www.youtube.com/embed/ykHiSn1K1E"
            }
          ]
        ],
        currentPlaylist: 0,
        currentVideo: 1,
        form: { artist: "", title: "", url: "" }
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });

  describe("when does not have playlists", () => {
    it("renders nothing", () => {
      const state = {
        playlists: null,
        currentPlaylist: null,
        currentVideo: null,
        form: { artist: "", title: "", url: "" }
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });
});
