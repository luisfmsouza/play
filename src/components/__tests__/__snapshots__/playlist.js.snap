// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist component when does not have playlists renders nothing 1`] = `null`;

exports[`Playlist component when there is playlists renders default component 1`] = `
Array [
  <h3
    className="sc-htpNat bpiAFK"
  >
    Playlists
  </h3>,
  <div
    className="sc-bdVaJa gexCij"
  >
    <button
      className="sc-gzVnrw hwdwgU"
      onClick={[Function]}
    >
      + add playlist
    </button>
    <div
      className="sc-bwzfXH iwEgFM"
    >
      <button
        className="sc-bxivhb jELbNG"
        onClick={[Function]}
      >
        <img
          className="sc-ifAKCX fPFXNf"
          src="https://placekitten.com/210/118"
        />
      </button>
      <button
        className="sc-bxivhb jELbNG"
        onClick={[Function]}
      >
        <img
          className="sc-ifAKCX fPFXNf"
          src="https://placekitten.com/210/118"
        />
        <div
          className="sc-EHOje tIvPg"
        >
          <span
            className="sc-bZQynM glHohg"
          >
            Play 
            0
             videos
          </span>
        </div>
      </button>
      <button
        className="sc-bxivhb jELbNG"
        onClick={[Function]}
      >
        <img
          className="sc-ifAKCX fPFXNf"
          src="https://placekitten.com/210/118"
        />
        <div
          className="sc-EHOje tIvPg"
        >
          <span
            className="sc-bZQynM glHohg"
          >
            Play 
            3
             videos
          </span>
        </div>
      </button>
      <button
        className="sc-bxivhb jELbNG"
        onClick={[Function]}
      >
        <img
          className="sc-ifAKCX fPFXNf"
          src="https://placekitten.com/210/118"
        />
        <div
          className="sc-EHOje tIvPg"
        >
          <span
            className="sc-bZQynM glHohg"
          >
            Play 
            1
             videos
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;
